### PR TITLE
connects Falkonry/insight#3691

### PR DESCRIPTION
**Pull request by phagunbaya**

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -320,7 +318,63 @@ Falkonry.prototype.addVerificationFromStream = function(pipeline, dataType, data
...
+  });
+};
+
+/**
+ *
+ * getOutputStream
+ *  To get output of a Pipeline
+ *
+ * @param {String} pipeline
+ * @param {Object} start
```
Comment (aniket-amrutkar) [inline lib/index.js]: is start Object ?

File: lib/service/falkonry.js [hunk 2]
```diff
@@ -263,6 +264,110 @@ FalkonryService.prototype.getOuptut = function(pipeline, start, end, done) {
 };
 
 /**
+ * To get a continuous stream of output data
+ * @param {String} pipeline
+ * @param {Object} start
```
Comment (aniket-amrutkar) [inline lib/service/falkonry.js]: same here

File: lib/service/falkonry.js [hunk 3]
```diff
@@ -263,6 +264,110 @@ FalkonryService.prototype.getOuptut = function(pipeline, start, end, done) {
...
+ * @param {Function} done
+ * @return {*}
+ */
+
+FalkonryService.prototype.streamOutput = function (pipeline, start, done) {
+  var _this = this;
+  var Pipeline = Models.Pipeline;
+  var path = '/pipeline/' + pipeline;
+  var url = '/Pipeline/' + pipeline +'/output?startTime='+start;
```
Comment (aniket-amrutkar) [inline lib/service/falkonry.js]: what if there is no start time ?

File: lib/service/falkonry.js [hunk 4]
```diff
@@ -263,6 +264,110 @@ FalkonryService.prototype.getOuptut = function(pipeline, start, end, done) {
...
+            awaitingResponse= false;
+            error = "Pipeline closed";
+            return outflowUpdate.emit('error', error);
+          }
+          else {
+            var request = _this.http.downstream(url);
+            var result = "";
+            //console.log("Request Headers : " + request/*getAllResponseHeaders().getResponseHeader("x-falkonry-data-offset")*/);
+            request.on('response', function (response) {
+              //console.log("ResponseBody : " + response.body);// + request.getResponseHeader("x-falkonry-data-offset"));
```
Comment (aniket-amrutkar) [inline lib/service/falkonry.js]: remove commneted code

File: README.md [hunk 5]
```diff
@@ -23,65 +23,139 @@ $ npm install falkonry-js-client
...
-var eventbuffer = new Schemas.Eventbuffer();
-eventbuffer.setName('Test-Eeventbuffer-01');
-eventbuffer.setTimeIdentifier("time");
-eventbuffer.setTimeFormat("iso_8601");
+//instantiate Falkonry
+var falkonry   = new Falkonry('https://service.falkonry.io', 'auth-token'); 
 
-return falkonry.createEventbuffer(eventbuffer, function(error, response){});
+var eventbuffer = new Schemas.Eventbuffer();    
+eventbuffer.setName('Test-Eeventbuffer-01');    //name of the eventbuffer
```
Comment (phagunbaya) [inline README.md]: Typo

File: README.md [hunk 6]
```diff
@@ -23,65 +23,139 @@ $ npm install falkonry-js-client
...
+//instantiate Falkonry
 var falkonry   = new Falkonry('https://service.falkonry.io', 'auth-token');
 
 var eventbuffer = new Schemas.Eventbuffer();
-eventbuffer.setName('Test-Eeventbuffer-01');
-eventbuffer.setTimeIdentifier("time");
-eventbuffer.setTimeFormat("iso_8601");
-eventbuffer.setThingIdentifier("motor");
-
+eventbuffer.setName('Test-Eeventbuffer-01');    //name of the eventbuffer
```
Comment (phagunbaya) [inline README.md]: typo

File: README.md [hunk 7]
```diff
@@ -23,65 +23,139 @@ $ npm install falkonry-js-client
...
-eventbuffer.setTimeFormat("YYYY-MM-DD HH:mm:ss");
-eventbuffer.setSignalsTagField("tag");
-eventbuffer.setSignalsDelimiter("_");
-eventbuffer.setSignalsLocation("prefix");
-eventbuffer.setValueColumn("value");
+//instantiate Falkonry
+var falkonry   = new Falkonry('https://service.falkonry.io', 'auth-token'); 
+
+var eventbuffer = new Schemas.Eventbuffer();    
+eventbuffer.setName('Test-Eeventbuffer-01');    //name of the eventbuffer
```
Comment (phagunbaya) [inline README.md]: typo

**Comment by phagunbaya** [comment]
Readme is not aligned with the Readme of java client. Please take that as a reference in terms of heading.

File: README.md [hunk 8]
```diff
@@ -23,105 +23,241 @@ $ npm install falkonry-js-client
...
 
 var eventbuffer = new Schemas.Eventbuffer();
-eventbuffer.setName('Test-Eeventbuffer-01');
-eventbuffer.setTimeIdentifier("time");
-eventbuffer.setTimeFormat("iso_8601");
+eventbuffer.setName('Test-Eeventbuffer-01');    //name of the eventbuffer
+eventbuffer.setTimeIdentifier("time");          //property that identifies time in the data
+eventbuffer.setTimeFormat("iso_8601");          //format of the time in the data
+eventbuffer.setSignalsTagField("tag");          //property that identifies signal tag in the data
+eventbuffer.setSignalsDelimiter("_");           //delimiter used to concat thing id and signal name to create signal tag
```
Comment (phagunbaya) [inline README.md]: For single thing, `signalDelimiter` and `signalLocation` is not required.

**Pull request merged**